paperqa/llms.py

```diff
@@ -510,21 +518,37 @@ def get_litellm_retrying_config(timeout: float = 60.0) -> dict[str, Any]:
     return {"num_retries": 3, "timeout": timeout}
 
 
-class LiteLLMModel(LLMModel):
-    """A wrapper around the litellm library.
+class PassThroughRouter(litellm.Router):
+    """Router that is just a wrapper on LiteLLM's normal free functions."""
 
-    `config` should have two high level keys:
-        `model_list`: stores a list of all model configurations
-          (see https://docs.litellm.ai/docs/routing)
-        `router_kwargs`: kwargs for the Router class
-        `rate_limit`: (Optional) dictionary keyed by model group name
-            with values of type limits.RateLimitItem (in tokens / minute)
-            or valid limits.RateLimitItem string for parsing
+    def __init__(self, **kwargs):
+        self._default_kwargs = kwargs
 
-    This way users can specify routing strategies, retries, etc.
-    """
+    async def atext_completion(self, *args, **kwargs):
+        return await litellm.atext_completion(*args, **(self._default_kwargs | kwargs))
 
-    config: dict = Field(default_factory=dict)
+    async def acompletion(self, *args, **kwargs):
+        return await litellm.acompletion(*args, **(self._default_kwargs | kwargs))
+
+
+class LiteLLMModel(LLMModel):
+    """A wrapper around the litellm library."""
+
+    config: dict = Field(
+        default_factory=dict,
+        description=(
+            "Configuration of this model containing several important keys. The"
+            " optional `model_list` key stores a list of all model configurations"
+            " (SEE: https://docs.litellm.ai/docs/routing). The optional"
+            " `router_kwargs` key is keyword arguments to pass to the Router class."
+            " Inclusion of a key `pass_through_router` with a truthy value will lead"
+            " to using not using LiteLLM's Router, instead just LiteLLM's free"
+            f" functions (see {PassThroughRouter.__name__}). The optional `rate_limit`"
```

Comment:
We might note that the rate limiter will only work if using the router based config.

Reply:
Nice suggestion, added a note here about this. We discussed in person, but rate limits apply to both cases